opampserver/pkg/connection/conncache.go

```diff
@@ -142,3 +143,12 @@ func (c *ConnectionsCache) UpdateAllConnectionConfigs(connConfigEvaluator func(c
 		}
 	}
 }
+
+func (c *ConnectionsCache) GetConnectionInfoByWorkload(podWorkload workload.PodWorkload) (*ConnectionInfo, error) {
```

Comment:
we can have multiple connections for the same workload, right? should this be a slice?